src/util/apiUtils.js

```diff
@@ -82,14 +82,27 @@ export const parseApiResponse = requestUrl => ([response, body]) => {
 		};
 	}
 	try {
-		value = JSON.parse(body);
-		if (value && value.problem) {
-			value = formatApiError(new Error(`API problem: ${value.problem}: ${value.details}`));
+		if (response.statusCode === 204) { //NoContent response type
+			value = {};
+			const successMeta = {
+				success: true,
+				...meta
+			};
+
+			return {
+				value,
+				meta: successMeta,
+			};
+		} else {
+			value = JSON.parse(body);
+			if (value && value.problem) {
+				value = formatApiError(new Error(`API problem: ${value.problem}: ${value.details}`));
+			}
+			return {
+				value,
+				meta,
+			};
```

Comment:
This is getting more nested than I'd like - there are a lot of ways of exiting the function, which might make it hard to debug. I think we can simplify by taking a step back.

1. do we need a `meta.success` prop, or can we assume a "`value`-with-no-`error`-prop" is a success?
2. if the underlying issue is that we're trying to `JSON.parse` an empty value, why not just skip `JSON.parse` when the value is empty?

would this simple change address the problem?

```js
value = body && JSON.parse(body) || {};
```